### DISCUSSION TITLE
[Chapter 04] 적절한 REST-API 설계란 무엇일까요?

### DISCUSSION THREAD

**Discussion by Irisation23** [cowaine/spring-boot-for-msa#19] [Chapter 04. Spring Web MVC Overview] [answered]
### `4.3 REST-API 설계` 

위의 주제에 해당 되는 글을 읽어보면 아래의 구문을 볼 수 있습니다.

```
일관성이 없는 REST-API는 개발보다 다른 일에 더 많은 시간이 필요하다.
```

1. 여기서 말하는 `일관성`을 잘 따르고 있는 좋은 예의 API가 있을까요?
2. API의 설계는 `리소스`, `행위`, `표현` 을 기본으로 하여 작성 되는데 이외에 고려할 점은 어떤 것이 있을까요?

**Comment by corock** [comment, marked as answer]
1. 저는 [토스페이먼츠 API](https://docs.tosspayments.com/reference) 가 제일 _best practice_ 였습니다.
일관성이라는 게 조직이나 프로젝트 규모가 클수록 잘 유지하기가 쉽지 않은 것 같네요.
어쩌면 그 일관성을 잘 유지할 수 있도록 하는 게 PM or PL 의 역할이 아닐까 싶기도...

2. 개인적으로 하나 꼽자면 API 를 설계할 때 설계자가 [케밥 케이스](https://namu.wiki/w/%EC%BD%94%EB%94%A9%20%EC%8A%A4%ED%83%80%EC%9D%BC#s-3.4)를 잘 지켜줬으면 하는 바람이 있습니다.

**Comment by Irisation23** [comment]
`URL` 과 `URI`의 차이는 뭘까요?

//TODO
